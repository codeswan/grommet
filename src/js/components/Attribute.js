// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React, { Component, PropTypes } from 'react';
import CSSClassnames from '../utils/CSSClassnames';

const CLASS_ROOT = CSSClassnames.ATTRIBUTE;

export default class Attribute extends Component {
  render () {
    return (
      <div className={CLASS_ROOT}>
        <label className={`${CLASS_ROOT}__label`}>
          {this.props.label}
        </label>
        <span className={`${CLASS_ROOT}__contents`}>
          {this.props.children}
        </span>
      </div>
    );
  }
};

Attribute.propTypes = {
  label: PropTypes.string
};
