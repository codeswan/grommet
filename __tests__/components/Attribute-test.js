// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Attribute from '../../src/js/components/Attribute';

describe('Attribute', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Attribute label='testLabel'>testContent</Attribute>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
