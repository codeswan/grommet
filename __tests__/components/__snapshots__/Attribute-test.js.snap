exports[`Attribute has correct default options 1`] = `
<div
  className="grommetux-attribute">
  <label
    className="grommetux-attribute__label">
    testLabel
  </label>
  <span
    className="grommetux-attribute__contents">
    testContent
  </span>
</div>
`;
